### PR TITLE
feat: Add RBAC API for GetAllDomains

### LINKED ISSUE #971: [Feature] I want to get the current domains
t seems that there is no API for getdomains. Would you like to add it? Because the adapter is implemented with MySQL

### PR DESCRIPTION
**Pull request by BudhirajaMadhav**
Fixes #971 
Adds an API to get all the domains.

Kindly review @hsluoyz and please tell if you have any suggestions.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@BudhirajaMadhav fix:

image

File: rbac_api_with_domains_test.go [hunk 1]
```diff
@@ -250,3 +250,14 @@ func TestDeleteAllUsersByDomain(t *testing.T) {
 		{"alice", "admin", "domain1"},
 	})
 }
+
+// testGetAllDomains tests GetAllDomains()
+func testGetAllDomains(t *testing.T, e *Enforcer, res []string) {
```
Comment (sagilio) [inline rbac_api_with_domains_test.go:255]: Are there some tests called this func?

File: rbac_api_with_domains.go [hunk 2]
```diff
@@ -137,3 +137,8 @@ func (e *Enforcer) DeleteDomains(domains ...string) (bool, error) {
 	}
 	return true, nil
 }
+
+// GetAllDomains would get all domains.
+func (e *Enforcer) GetAllDomains() ([]string, error) {
+	return e.model["p"]["p"].RM.GetAllDomains()
```
Comment (sagilio) [inline rbac_api_with_domains.go:143]: We should use the `RM` in `g` section and `gX`  (like `g`, `g2`., `g3`..)

**Review by sagilio — CHANGES_REQUESTED** [review]
There are a few issues that need to confirm.

**Comment by wangdyqxx** [comment]
@BudhirajaMadhav  when is the merger expected to succeed ?

**Comment by hsluoyz** [comment]
@BudhirajaMadhav plz fix all CI errors:

image

**Comment by BudhirajaMadhav** [comment]
Hey @hsluoyz @wangdyqxx @sagilio I've fixed the CI tests and added the added the test. Please have a review.

File: rbac_api_with_domains.go (hunk 2 above)
Comment (BudhirajaMadhav) [reply rbac_api_with_domains.go]: Hi @sagilio, I changed the "p" section to "g", but can you please mention why we did that or can give me a resource to read about that?

**Comment by hsluoyz** [comment]
@BudhirajaMadhav plz answer all the review questions like:

image

File: rbac_api_with_domains_test.go (hunk 1 above)
Comment (BudhirajaMadhav) [reply rbac_api_with_domains_test.go:255]: I've added the test to call this function.

**Comment by BudhirajaMadhav** [comment]
Hey @hsluoyz, I've answered them now but can you please resolve [this](https://github.com/casbin/casbin/pull/974#discussion_r834661258) doubt of mine?

File: rbac_api_with_domains.go (hunk 2 above)
Comment (hsluoyz) [reply rbac_api_with_domains.go]: @BudhirajaMadhav see: https://casbin.org/docs/en/role-managers

**Comment by hsluoyz** [comment]
@BudhirajaMadhav replied

**Comment by BudhirajaMadhav** [comment]
Thanks @hsluoyz, if everything alright can we merge this request?

**Comment by hsluoyz** [comment]
@BudhirajaMadhav plz run `go fmt` for all your code

**Comment by hsluoyz** [comment]
@wangdyqxx @tangyang9464 plz review

**Comment by BudhirajaMadhav** [comment]
> @BudhirajaMadhav plz run `go fmt` for all your code

Sure. One more thing I wanted to ask, do you squash commits while merging or do I have to do it myself?

**Comment by tangyang9464** [comment]
Just run ``go fmt`` and it‘s ok.

**Review by tangyang9464 — APPROVED** [review]
Just run go fmt and it‘s ok.

**Comment by tangyang9464** [comment]
@BudhirajaMadhav You need to squash your commits.

**Comment by wangdyqxx** [comment]
no problem

**Comment by hsluoyz** [comment]
@BudhirajaMadhav plz also add this new API to docs: https://github.com/casbin/casbin-website/blob/master/docs/RBACAPI.md

**Pull request merged**